tests/src/Interop/PInvoke/FixedBuffer/MarshalFixedBuffer.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+using TestLibrary;
+
+public unsafe class Managed
+{
+    private const int BufferSize = 10;
+
+    struct BlittableFixedBuffer
+    {
+        public fixed int elements[BufferSize];
+    }
+
+    struct NonBlittableFixedBuffer
+    {
+        public fixed bool elements[BufferSize];
+    }
+
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+    struct ElementAfterNonBlittableFixedBuffer
+    {
+        public fixed char fixedString[BufferSize];
+        public int testValue;
+    }
+
+    [DllImport("MarshalFixedBuffer")]
+    private static extern int SumBlittableFixedBuffer(BlittableFixedBuffer buffer);
+
+    [DllImport("MarshalFixedBuffer")]
+    private static extern bool AggregateNonBlittableFixedBuffer(NonBlittableFixedBuffer buffer);
+
+    [DllImport("MarshalFixedBuffer")]
+    private static extern bool VerifyElementAfterNonBlittableFixedBuffer(ElementAfterNonBlittableFixedBuffer buffer);
+
+    private static ElementAfterNonBlittableFixedBuffer InitElementAfterNonBlittableFixedBuffer()
+    {
+        return new ElementAfterNonBlittableFixedBuffer
+        {
+            testValue = 42
+        };
+    }
+
+    private static void TestBlittableFixedBuffer()
+    {
+        Console.WriteLine("Start blittable fixed buffer test");
+
+        var buffer = new BlittableFixedBuffer();
+
+        for (int i = 0; i < BufferSize; i++)
+        {
+            buffer.elements[i] = i;
+        }
+
+        Assert.AreEqual(45, SumBlittableFixedBuffer(buffer));
```

Comment:
Boo. Tests should always compute 'expected' values if possible. In this case, perform the summation logic in the loop above.